### PR TITLE
Support barcode (kind of) in web apps

### PR DESCRIPTION
**Pull request by benrudolph**
@czue shows text entry for barcode question type. @wpride it needs this in order to work: https://github.com/dimagi/commcare-core/commit/723b22eed30bbab5fe249f57b65991aa2818fafb (couldn't get it into submodule branch, that branch seems to have changed)

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js [hunk 1]
```diff
@@ -163,7 +166,15 @@ function FreeTextEntry(question, options) {
     };
 
     self.helpText = function() {
-        return isPassword ? 'Password' : 'Free response';
+        if (isPassword) {
+            return gettext('Password');
+        }
+        switch (self.datatype) {
+        case Formplayer.Const.BARCODE:
```
Comment (czue) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js]: nit but is this intentionally not indented?

**Review by czue — APPROVED** [review]
nice - glad it was as easy as I thought. wasn't sure if this should have a "hold off" tag based on the commcare-core change so not merging just in case

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js (hunk 1 above)
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js]: yeah it's the way dimagimon likes it (that's why it yelled at the one above, that switch statement was written before we had JS lint)

**Comment by benrudolph** [comment]
yep needs that commcare-core commit before it'll work. i'll add tag

**Comment by wpride** [comment]
@benrudolph subrepo update here https://github.com/dimagi/formplayer/pull/196

branch in commcare-core is still "nimbus", might be good to add comment explaining why BARCODE is a string

**Comment by benrudolph** [comment]
oh nice! forgot that we're using nimbus branch. will update with comment

**Comment by wpride** [comment]
Gtm when tag removed

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js [hunk 2]
```diff
@@ -519,47 +530,49 @@ function getEntry(question) {
...
+        entry = new DateTimeEntry(question, {});
+        break;
+    case Formplayer.Const.GEO:
+        entry = new GeoPointEntry(question, {});
+        break;
+    case Formplayer.Const.INFO:
+        entry = new InfoEntry(question, {});
+        break;
+    default:
+        console.warn('No active entry for: ' + question.datatype());
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js]: Unexpected console statement. (no-console)

**Pull request merged**